orchestration/dagster_orchestration/hca_orchestration/contrib/data_repo.py

```diff
@@ -0,0 +1,45 @@
+import time
+import logging
+from typing import Callable, Any
+
+import data_repo_client
+
+
+class DataRepoApiRetryException(Exception):
+    """
+    Represents a failure after some number of retries against the data repo API
+    """
+    pass
+
+
+def api_request_with_retry(
+        api_fn: Callable[[Any], Any],
+        max_wait_time: int,
+        poll_interval_seconds: int,
+        retriable_status_codes: set[int],
+        *args: str) -> Any:
+    """
+    Polls on the given API call; any status codes return in the retriable_status_codes will not result in
+    immediate failure but instead cause a retry until max_wait_time is reached.
+    :param api_fn: API method to call
+    :param max_wait_time: Maximum wait time in seconds to poll
+    :param poll_interval_seconds: Interval between polling calls
+    :param retriable_status_codes:  Set of HTTP status codes that indicate we should retry
+    :param args: Args to pass to the API fn
+    :return:
+    """
+    time_waited = 0
+
+    while time_waited < max_wait_time:
```

Comment:
I would like to coalesce this and this [similar op](https://github.com/DataBiosphere/hca-ingest/blob/master/orchestration/dagster_orchestration/hca_orchestration/solids/data_repo.py#L22), but I am unsure atm how to integrate the exception handling needed here cleanly.